napari/components/_dims/view.py

```diff
@@ -1,89 +1,230 @@
+
+        Parameters
+        ----------
+        slider_index : slider index (corresponds to axis index)
+        """
+        if slider_index >= self.nsliders:
+            return
+
+        slider = self.sliders[slider_index]
+
+        if slider is None:
+            return
 
-    def _axis_to_row(self, axis, max_dims):
-        message = f'axis {axis} out of bounds for {max_dims} dims'
+        if slider_index < self.dims.ndims:
 
-        if axis < 0:
-            axis = max_dims - axis
-            if axis < 0:
-                raise IndexError(message)
-        elif axis >= max_dims:
-            raise IndexError(message)
+            mode = self.dims.mode[slider_index]
+            if mode == DimsMode.POINT:
+                slider.collapse()
+                slider.setValue(self.dims.point[slider_index])
+            elif mode == DimsMode.INTERVAL:
+                slider.expand()
+                slider.setValues(self.dims.interval[slider_index])
+            slider_range = self.dims.range[slider_index]
+            if slider_range not in (None, (None, None, None)):
+                slider.setRange(slider_range)
 
-        if axis < 2:
-            raise ValueError('cannot convert y/x-axes to rows')
+    def _update_nsliders(self):
+        """
+        Updates the number of sliders based on the number of dimensions
+        """
+        self._set_nsliders(self.dims.ndims-2)
 
-        return axis - 2
+    def _set_nsliders(self, new_number_of_sliders):
+        """
+        Sets the number of sliders displayed
+        Parameters
+        ----------
+        new_number_of_sliders :
+        """
+        if self.nsliders < new_number_of_sliders:
+            self._create_sliders(new_number_of_sliders)
+        elif self.nsliders > new_number_of_sliders:
+            self._trim_sliders(new_number_of_sliders)
 
-    def update_slider(self, event):
-        """Updates a slider for the given axis or creates
-        it if it does not already exist.
+    def _create_sliders(self, number_of_sliders):
+        """
+        Creates sliders to match new number of dimensions
+        Parameters
+        ----------
+        number_of_sliders : new number of sliders
+        """
+        while number_of_sliders > self.nsliders:
+            new_slider_axis = self.nsliders
+            slider = self._create_range_slider_widget(new_slider_axis)
+            for i in range(self.nsliders):
+                item = self.layout().takeAt(i)
+                self.layout().addWidget(item.widget(), i+1, 0)
+            self.layout().addWidget(slider, 0, 0)
+            self.sliders.append(slider)
+            self.setMinimumHeight(self.nsliders * self._slider_height)
 
+    def _trim_sliders(self, number_of_sliders):
+        """
+        Trims number of dimensions to a lower number
+        Parameters
+        ----------
+        number_of_sliders : new number of sliders
+        """
+        while number_of_sliders < self.nsliders:
```

Comment:
As above: use a suitable for-loop

Reply:
switched to `for` loop, also added `pop(0)` to trim from front of list

Reply:
actually `pop(0)` was bad so I have removed